### ISSUE TITLE
Memory error

### ISSUE THREAD

**Issue by tomasbruna** [lh3/miniprot#39]
Dear Heng,

I encountered the following memory error when running miniprot:

```
$ miniprot genome.fasta problem.fa 
[M::mp_ntseq_read@0.285*1.01] read 74282399 bases in 1 contigs
[M::mp_idx_build@0.285*1.01] 580332 blocks
[M::mp_idx_build@1.745*1.82] collected syncmers
[M::mp_idx_build@1.974*1.73] 29640897 kmer-block pairs
[M::mp_idx_print_stat] 1899129 distinct k-mers; mean occ of infrequent k-mers: 15.61; 0 frequent k-mers accounting for 0 occurrences
[kfree] The end of a free block enters the allocated block.
```

I tested on both Mac and Linux, both give the same error. Here are the genome and the problematic protein: https://drive.google.com/drive/folders/1O40t8TUzFBvhdFI8ZFRZugcoiEA29Vlw?usp=sharing

Best,
Tomas

**Comment by lh3** [comment]
@KatharinaHoff reported another error on #38. Possibly related. I will find time to fix both. Thank you and @KatharinaHoff a lot for providing the examples.

**Comment by lh3** [comment]
@KatharinaHoff and @tomasbruna: I believe I have fixed the bug. Once you can confirm on your side, I will cut a new release just for this fix. Thank you for the useful test cases.

**Comment by KatharinaHoff** [comment]
Dear Heng,
I confirm that my problem was fixed, no more segmentation fault on the data that I provided. Great! Thank you!
Best,
Katharina

**Comment by tomasbruna** [comment]
I confirm as well, this fixed the issue for me. Thanks!

**Comment by lh3** [comment]
Thanks for the confirmation. I have just released v0.10 for this bugfix – segfault is always bad.
